crates/l3/common/src/lib.rs

```
@@ -0,0 +1,53 @@
+pub mod errors;
+use starknet_accounts::{Account, Call, ConnectedAccount, Execution, SingleOwnerAccount};
+use starknet_core::types::{BlockId, BlockTag, FunctionCall};
+use starknet_ff::FieldElement;
+use starknet_providers::jsonrpc::{HttpTransport, JsonRpcClient};
+use starknet_providers::Provider;
+use starknet_signers::LocalWallet;
+use std::sync::Arc;
+
+use starknet_core::utils::get_selector_from_name;
+
+pub type LocalWalletSignerMiddleware =
+    SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>;
+pub const NO_CONSTRUCTOR_ARG: Vec<FieldElement> = Vec::new();
+
+pub trait StarknetContractClientL3 {
+    fn address(&self) -> FieldElement;
+    fn client(&self) -> LocalWalletSignerMiddleware;
+}
+
+pub async fn invoke_contract<'a>(
+    client: &'a Arc<LocalWalletSignerMiddleware>,
+    address: FieldElement,
+    method: &str,
+    calldata: Vec<FieldElement>,
+) -> Execution<'a, LocalWalletSignerMiddleware> {
+    let call = Call {
+        to: address,
+        selector: get_selector_from_name(method.into()).unwrap(),
```

we should throw the error instead on unwrap

---

done

---

i think this is still pending?